File: windows/src/ConnectorControl.Core/State/ArgRow.cs
namespace ConnectorControl.Core.State;

/// <summary>One argument text box; the row object's own identity keeps focus while the list is edited.</summary>
public sealed class ArgRow : ObservableObject
{
    private string value;

    public ArgRow(string value)
    {
        this.value = value;
    }

    public string Value { get => value; set => Set(ref this.value, value); }
}

File: Sources/ConnectorControlState/ArgRow.swift
import Foundation

/// One argument field; a stable identity keeps focus while the list is edited.
public struct ArgRow: Identifiable, Equatable, Sendable {
    public let id: UUID
    public var value: String

    public init(value: String) {
        id = UUID()
        self.value = value
    }
}
